From: Wang Nan <wangnan0@huawei.com>
Subject: memory-hotplug: ppc: suitable memory should go to ZONE_MOVABLE

References: fate#316483
Patch-mainline: No, expected 3.17

This patch introduces zone_for_memory() to arch_add_memory() on powerpc to
ensure new, higher memory added into ZONE_MOVABLE if movable zone has
already setup.

Signed-off-by: Wang Nan <wangnan0@huawei.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Mel Gorman <mgorman@suse.de>
---

 arch/powerpc/mm/mem.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff -puN arch/powerpc/mm/mem.c~memory-hotplug-ppc-suitable-memory-should-go-to-zone_movable arch/powerpc/mm/mem.c
--- a/arch/powerpc/mm/mem.c~memory-hotplug-ppc-suitable-memory-should-go-to-zone_movable
+++ a/arch/powerpc/mm/mem.c
@@ -128,7 +128,8 @@ int arch_add_memory(int nid, u64 start,
 		return -EINVAL;
 
 	/* this should work for most non-highmem platforms */
-	zone = pgdata->node_zones;
+	zone = pgdata->node_zones +
+		zone_for_memory(nid, start, size, 0);
 
 	return __add_pages(nid, zone, start_pfn, nr_pages);
 }
